examples/standard_v2_instance/main.tf

```diff
@@ -0,0 +1,118 @@
+  subnet_names                                         = var.subnet_names
+  subnet_prefixes                                      = var.subnet_prefixes
+  bgp_community                                        = null
+  ddos_protection_plan                                 = null
+  dns_servers                                          = []
+  nsg_ids                                              = {}
+  route_tables_ids                                     = {}
+  subnet_delegation                                    = {}
+  subnet_private_endpoint_network_policies_enabled     = {}
+  subnet_private_link_service_network_policies_enabled = {}
+  subnet_service_endpoints                             = {}
+  tags                                                 = merge(var.tags, { resource_name = module.resource_names["vnet"].standard })
+  use_for_each                                         = true
+
+
+  depends_on = [module.resource_group]
+}
+
+module "application_gateway" {
+  source = "../.."
+
+  name                                   = module.resource_names["app_gateway"].standard
+  location                               = var.region
+  resource_group_name                    = module.resource_group.name
+  frontend_ip_configuration_name         = var.frontend_ip_configuration_name
+  public_ip_address_id                   = module.public_ip.id
+  frontend_private_ip_configuration_name = var.frontend_private_ip_configuration_name
+  gateway_ip_configuration_name          = var.gateway_ip_configuration_name
+  sku                                    = var.sku
+  sku_capacity                           = var.sku_capacity
+  zones                                  = var.zones
+  frontend_port_settings                 = var.frontend_port_settings
+  trusted_client_certificates_configs    = var.trusted_client_certificates_configs
+  trusted_root_certificate_configs       = var.trusted_root_certificate_configs
+  appgw_backend_pools                    = var.appgw_backend_pools
+  appgw_http_listeners                   = var.appgw_http_listeners
+  appgw_routings                         = var.appgw_routings
+  appgw_probes                           = var.appgw_probes
+  appgw_backend_http_settings            = var.appgw_backend_http_settings
+  subnet_id                              = module.vnet.vnet_subnets[0]
+  user_assigned_identity_id              = var.user_assigned_identity_id
+  appgw_private                          = var.appgw_private
+  appgw_private_ip                       = var.private_ip_address
+  enable_http2                           = var.enable_http2
+  autoscaling_parameters                 = var.autoscaling_parameters
+  ssl_certificates_configs               = var.ssl_certificates_configs
+  authentication_certificates_configs    = var.authentication_certificates_configs
+
+  tags = merge(var.tags, {
+    resource_name = module.resource_names["app_gateway"].standard
```

Comment:
Once you address my comment in the root `locals.tf`, this additional merge of the resource_name tag should be unnecessary.

Reply:
resource_name = module.resource_names["app_gateway"].standard is must when we use any other format of the name other than `standard`. Per @matt-dresden-nttd the resource_name must always is using `standard`

Reply:
I respectfully disagree with the above guidance with regard to examples. For the record, I think that our reference modules _should_ pass the standard naming convention (where possible) to all the resources it creates. For the examples, this will create more problems than it solves.

Concurrent test runs (e.g. two people working on different branches of the same module at the same time) are not possible with the standard naming convention, because as person A’s tests may succeed, person B’s tests target resources with a conflicting name. The best case scenario is that tests fail cleanly due to a naming conflict, there are several progressively worse scenarios where tests partially complete due to a race in resource availability (person A’s resources are being torn down as person B’s tests are executing), or a false positive result in testing where a true negative should have been caught.

There are two fundamental problems with the requirement that the resource_name tag contain the standard naming convention variant and only that value:

1. If our standard name can’t be used with the resource in question (storage accounts are the prime example here) then we will have a mismatch between the resource’s name and the resource_name tag.

2. Primitive modules by design do not include the resource_name module because they’re not intended to figure out their own names — they’re intended to have their names supplied — but if there’s a need to apply the standard naming convention to one of the tags involved, that means that **every higher order module that contains a primitive that supports tags will need to be updated to pass the standard naming convention value in as well as the actual name of the resource**! This is an enormous amount of boilerplate to support tagging our primitive modules with a name that might not actually reflect reality. The alternative is that we have to include resource_names on each of the primitives which is a similarly massive undertaking, because we're not just passing the name anymore, we'd have to pass the full logical family, logical product, class env, etc. set of options to the primitive so it can construct its name... not ideal!

Reply:
After thinking on this a little more, litigating this in a PR review probably isn't the right answer. Let's see if we can get some time on Matt's calendar to discuss in more depth, I suspect we can find a solution that meets everyone's needs.